src/window.rs

```diff
@@ -257,19 +257,34 @@ impl WindowUtil for Window {
             let k = (*consumed % WINDOW_SIZE) as usize;
             trace!("{}: k: {} consumed: {}", id, k, *consumed,);
 
-            if let Some(blob) = &self[k].data {
+            let k_data_blob;
+            let k_data_slot = &mut self[k].data;
+            if let Some(blob) = k_data_slot {
                 if blob.read().get_index().unwrap() < *consumed {
                     // window wrap-around, end of received
                     break;
                 }
+                k_data_blob = (*blob).clone();
             } else {
                 // self[k].data is None, end of received
                 break;
             }
-            let slot = self[k].clone();
-            if let Some(r) = slot.data {
-                consume_queue.push(r)
+
+            // Check that we can get the entries from this blob
+            match reconstruct_entries_from_blobs(vec![k_data_blob]) {
+                Ok(entries) => {
+                    consume_queue.extend(entries);
+                }
+                Err(_) => {
+                    // If the blob can't be deserialized, then remove it from the
+                    // window and exit. *k_data_slot cannot be None at this point,
+                    // so it's safe to unwrap.
+                    let old = mem::replace(k_data_slot, None).unwrap();
+                    recycler.recycle(old, "insert_blob_is_dup");
```

Comment:
Can you rebase on #1290 after it's merged?

Reply:
sure!